### DISCUSSION TITLE
Schema help: User created types/ values

### DISCUSSION THREAD

**Discussion by JeffersonBledsoe** [prisma/prisma#14592] [Q&A] [answered]
Hi all. I'm working on an app where the user can add an attribute to an object. They select the name of the attribute, what kind of attribute it is (e.g. a data, a number, etc.) and then the value is calculated based on the type of attribute. Currently my testing schema (only string and date) looks like this:

```prisma
model Object {
  data      Attribute[]
}

enum AttributeTypes {
  string
  date
}

model Attribute {
  attributeName String         @unique
  type          AttributeTypes
  value         String // Probably a better type here
  Object         Object?         @relation(fields: [objectId], references: [id])
  objectId       String?
}
```

Is there any way to make this stricter/ represent the types better rather than having to convert the attribute type to the actual type in the client?

Fairly new to Prisma, but loving it!

**Comment by austincrim** [comment, marked as answer]
Hey @JeffersonBledsoe 👋,

I don't think there is a "stricter" or more correct way to accomplish this. Without knowing the type of the value ahead of time, `String` is a good general container to hold other values.

You could take a look at [writing a middleware](https://www.prisma.io/docs/concepts/components/prisma-client/middleware) to handle the parsing of the value for you.

**Reply by JeffersonBledsoe** [reply]
Hey, thanks for the reply! I'll look into middleware 🙂

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
